packages/node_modules/@node-red/nodes/core/function/10-function.js

```diff
@@ -415,8 +417,15 @@ module.exports = function(RED) {
                     var opts = {};
                     if (node.timeout>0){
                         opts = node.timeoutOptions;
+                    } else if(RED.settings.defaultFunctionTimeout > 0){
+                        opts.timeout = RED.settings.defaultFunctionTimeout * 1000
+                    }
+                    try {
+                        node.script.runInContext(context,opts);
+                    } catch (err) {
+                        node.error(err);
+                        return done(err);
```

Comment:
```suggestion
                        return done(err);
```

Calling `done(err)` will do the right thing with the logging